packages/mysql/lib/mysql_p.d.ts

```diff
@@ -72,4 +74,69 @@ declare namespace captureMySQL {
     ? PatchedCreatePoolClusterFunction
     : T[K];
   };
+
+  type MySQL2QueryCallbackResultArg =
+  | MySQL2.RowDataPacket[][]
+  | MySQL2.RowDataPacket[]
+  | MySQL2.OkPacket
+  | MySQL2.OkPacket[]
+  | MySQL2.ResultSetHeader;
+
+  type MySQL2QueryCallback = <T extends MySQL2QueryCallbackResultArg>(
+      err: MySQL2.QueryError | null,
+      result: T,
+      fields: MySQL2.FieldPacket[],
+    ) => any;
+
+  interface PatchedMySQL2PoolGetConnectionFunction {
+    (): Promise<PatchedConnection>;
+  }
+
+  type PatchedMySQL2Pool<T = MySQL2.Pool> = {
+    [K in keyof T]: K extends 'query'
+    ? PatchedQueryFunction
+    : K extends 'getConnection'
+    ? PatchedMySQL2PoolGetConnectionFunction
+    : T[K];
+  };
+
+  interface PatchedMySQL2PoolClusterOfFunction {
+    (pattern: string, selector?: string): PatchedMySQL2Pool;
+  }
+
+  interface PatchedMySQL2PoolClusterGetConnectionFunction {
+    (): Promise<PatchedConnection>;
+    (group: string): Promise<PatchedConnection>;
+    (group: string, selector: string): Promise<PatchedConnection>;
+  }
+
+  type PatchedMySQL2PoolCluster<T = MySQL.PoolCluster> = {
```

Comment:
Should this be `MySQL2.PoolCluster`?

Reply:
I'll take a look tomorrow and get back to you, thanks!

Reply:
Good spot, you're right - that's amended now. [3050e6d](https://github.com/aws/aws-xray-sdk-node/pull/676/commits/3050e6d31e237b6d76da5d04a39fb49e291b9432)